util/types/datum.go

```
@@ -1309,14 +1276,13 @@ func (d *Datum) ToBool(sc *variable.StatementContext) (int64, error) {
 	case KindMysqlDecimal:
 		v, _ := d.GetMysqlDecimal().ToFloat64()
 		isZero = (RoundFloat(v) == 0)
-	case KindMysqlHex:
-		isZero = (d.GetMysqlHex().ToNumber() == 0)
-	case KindMysqlBit:
-		isZero = (d.GetMysqlBit().ToNumber() == 0)
 	case KindMysqlEnum:
 		isZero = (d.GetMysqlEnum().ToNumber() == 0)
 	case KindMysqlSet:
 		isZero = (d.GetMysqlSet().ToNumber() == 0)
+	case KindBinaryLiteral, KindMysqlBit:
+		val, _ := d.GetBinaryLiteral().ToInt()
```

error is not handled.